src/server/JasmineGraphServer.cpp

```diff
@@ -1319,697 +1129,126 @@ void JasmineGraphServer::copyCentralStoreToAggregateLocation(std::string filePat
-        server_logger.log("Received : " + JasmineGraphInstanceProtocol::HANDSHAKE_OK, "info");
-        result_wr = write(sockfd, masterIP.c_str(), masterIP.size());
+    while (true) {
+        unsigned char buff[1024] = {0};
+        int nread = fread(buff, 1, 1024, fp);
 
-        if (result_wr < 0) {
-            server_logger.log("Error writing to socket", "error");
+        /* If read was success, send data. */
+        if (nread > 0) {
+            write(sockfd, buff, nread);
         }
 
-        server_logger.log("Sent : " + masterIP, "info");
-        bzero(data, FED_DATA_LENGTH + 1);
-        read(sockfd, data, FED_DATA_LENGTH);
-        response = (data);
-
-        if (response.compare(JasmineGraphInstanceProtocol::HOST_OK) == 0) {
-            server_logger.log("Received : " + JasmineGraphInstanceProtocol::HOST_OK, "info");
-        } else {
-            server_logger.log("Received : " + response, "error");
+        if (nread < 1024) {
+            if (feof(fp)) printf("End of file\n");
+            if (ferror(fp)) printf("Error reading\n");
+            break;
         }
+    }
 
-        result_wr = write(sockfd, JasmineGraphInstanceProtocol::UPLOAD_RDF_ATTRIBUTES.c_str(),
-                          JasmineGraphInstanceProtocol::UPLOAD_RDF_ATTRIBUTES.size());
+    fclose(fp);
+    close(sockfd);
+    return true;
+}
 
-        if (result_wr < 0) {
-            server_logger.log("Error writing to socket", "error");
-        }
+static void copyArtifactsToWorkers(const std::string &workerPath, const std::string &artifactLocation,
+                                   const std::string &remoteWorker) {
+    if (artifactLocation.empty() || artifactLocation.find_first_not_of(' ') == artifactLocation.npos) {
+        server_logger.error("Received `" + artifactLocation + "` for `artifactLocation` value!");
+        throw std::invalid_argument("Received empty string for `artifactLocation` value!");
+    }
 
-        server_logger.log("Sent : " + JasmineGraphInstanceProtocol::UPLOAD_RDF_ATTRIBUTES, "info");
-        bzero(data, FED_DATA_LENGTH + 1);
-        read(sockfd, data, FED_DATA_LENGTH);
-        response = (data);
-        response = Utils::trim_copy(response, " \f\n\r\t\v");
-        // std::cout << response << std::endl;
-        if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
-            server_logger.log("Received : " + JasmineGraphInstanceProtocol::OK, "info");
-            // std::cout << graphID << std::endl;
-            result_wr = write(sockfd, std::to_string(graphID).c_str(), std::to_string(graphID).size());
-
-            if (result_wr < 0) {
-                server_logger.log("Error writing to socket", "error");
-            }
+    std::string pathCheckCommand = "test -e " + workerPath;
+    if (remoteWorker.find("localhost") == std::string::npos) {
+        pathCheckCommand = "ssh -p 22 " + remoteWorker + " " + pathCheckCommand;
```

Comment:
Shall we make SSH port a configurable and read it from `jasminegraph-server.properties` file? I think keeping it configurable would be useful in certain adverse scenarios. See https://linuxhandbook.com/ssh-alternate-port/

Reply:
It's not necessary to set `-p 22`. It's the default anyway. Instead of modifying the command, we can set it from `.ssh/config`

Reply:
Ok